### PR TITLE
[ImageCache] don't convert CYMK image to RGB

### PR DESCRIPTION
**Pull request by troopa81**
This PR is related to [CMYK QEP](https://github.com/qgis/QGIS-Enhancement-Proposals/issues/283)

CMYK image support has been added in Qt 6.8.0. For now, it's not possible to render in a CMYK image, and as a consequence to scale it. So, QGIS will report a warning message to user if requested size is different from the CMYK image one.

**Funded by Métropôle de Bordeaux**

### CODE REVIEW

**Comment by nyalldawson** [comment]
> For now, it's not possible to render in a CMYK image, and as a consequence to scale it

Out of interest, is addressing this limitation on kdab's roadmap? From a niave pov I can't see why scaling a CMYK image is any different then scaling a rgba one

File: src/core/qgsimagecache.cpp [hunk 1]
```diff
@@ -491,6 +495,13 @@ QImage QgsImageCache::renderImage( const QString &path, QSize size, const bool k
   // render image at desired size -- null size means original size
   if ( !size.isValid() || size.isNull() || im.size() == size )
     return im;
+#if QT_VERSION >= QT_VERSION_CHECK(6, 8, 0)
+  else if ( im.format() == QImage::Format_CMYK8888 )
+  {
+    QgsMessageLog::logMessage( tr( "Impossible to scale a CMYK image : %1" ).arg( path ) );
+    return im;
```
Comment (nyalldawson) [inline src/core/qgsimagecache.cpp:502]: From a practical perspective, I don't think returning the unscaled image is very useful here. The result is going to be unusable in almost every case. Maybe we should force convert to RGB and then scale?

**Comment by troopa81** [comment]
> Out of interest, is addressing this limitation on kdab's roadmap? From a niave pov I can't see why scaling a CMYK image is any different then scaling a rgba one

Quoting KDAB developer own words

> * Then the second step would be to support a minimal set of operations
  that make sense and are relatively easy to add: cropping, mirroring,
  90° rotations, and possibly a few others. Scaling is possibly
  already tricky to do correctly.

I take a look on qimage.cpp transform method (in charge of scaling) and there is two way of scaling
- calling [smoothscaled](https://codebrowser.dev/qt6/qtbase/src/gui/image/qimage.cpp.html#4854) method
- [painting](https://codebrowser.dev/qt6/qtbase/src/gui/image/qimage.cpp.html#4925) with a transform matrix

The second wouldn't work as painting on CMYK image is not possible. Regarding the first, CMYK8888 format is not among the smoothscaled allowed formats, so that's maybe an oversight (I'm gonna reach out KDAB developer regarding this). Funny thing is I can force call of smoothscaled method by asking a [2 times smaller scale](https://codebrowser.dev/qt6/qtbase/src/gui/image/qimage.cpp.html#4820) and it works without error.

Anyway, long story short, we cannot make assumption of what Qt would return when we scale an image, and there is no way to detect if scaling has worked or not (except for the message `QPainter::begin: Cannot paint on an image with the QImage::Format_CMYK8888 format` that would pop-up). So I propose to get rid of the QgsMessageLog part. What do you think?

File: src/core/qgsimagecache.cpp (hunk 1 above)
Comment (troopa81) [reply src/core/qgsimagecache.cpp:502]: Regarding my [comment](https://github.com/qgis/QGIS/pull/57417#issuecomment-2111783411), I propose to remove this

Comment (nyalldawson) [reply src/core/qgsimagecache.cpp:502]: Yep, +1 to remove

**Review by nyalldawson — APPROVED** [review]
